### PR TITLE
[meshtastic][board] add Happymodel ES900TX board

### LINKED ISSUE #2065: [Board]: Happymodel ES900TX
### SOC

ESP32

### Lora IC

SX1276

### Product Link

https://www.happymodel.cn/index.php/2021/02/19/expresslrs-module-es915tx-long-range-915mhz-transmitter-and-es915rx-receiver/

### Description

This model has onboard PA which can produce up to 2W radio output, price is pretty cheap for this power.
Disadvantages - no I2C pins easily accessible, no extra serial pins easily accessible.
However in the future one make use of ES8285 neighbor chip.

### PR DESCRIPTION
**Pull request by lesykm**
!!!DO NOT ACCEPT YET, POC, PROTOBUFS PR INCOMING UPON DISCUSSION FINISHED

fixes #2065

### CODE REVIEW

**Comment by lesykm** [comment]
Boot log: https://pastebin.com/nwVe1HAE

**Comment by mc-hamster** [comment]
Converted to draft

File: variants/happymodel_es900tx/variant.h [hunk 1]
```diff
@@ -0,0 +1,56 @@
+// This board do not have extra serial to connect GPS to
+#undef GPS_RX_PIN
+#undef GPS_TX_PIN
+
+// This board do not have easy accessible I2C pins
+#undef I2C_SDA
+#undef I2C_SCL
+
```
Comment (caveman99) [inline variants/happymodel_es900tx/variant.h:8]: This just means the firmware will use the standard 21/22 for I2C - if that's not occupied its fine, else please move them out of the way.

**Review by caveman99 — APPROVED** [review]

**Comment by caveman99** [comment]
@lesykm any intention to finish and stress test this? we have a PR for the sister board (2.4 GHz) as well. Maybe we can sync those 2 PR's up?

**Comment by garthvh** [comment]
Does this use the same problematic ESP32 chip that is on the heltec v1 and V2?

**Comment by sachaw** [comment]
Any update on this, are we still waiting on protobufs?

**Comment by geeksville** [comment]
This pull request has been mentioned on **Meshtastic**. There might be relevant details there:

https://meshtastic.discourse.group/t/add-wip-support-for-betafpv-elrs-micro-tx-module-2-4g-500-1000-mw-with-esp32-cpu/7023/12

**Comment by garthvh** [comment]
Is this board working well, or have there been issues since it booted?

**Comment by lesykm** [comment]
Sorry, no time to finish, anyone may use this as a reference.

**Comment by uberhalit** [comment]
> Sorry, no time to finish, anyone may use this as a reference.
@lesykm do you remember what parts are missing for a functional release? I got a RadioMaster Bandit Nano which uses the same ESP32-D0WDQ6 + SX1276 combo as your TX. I ported over some code from ELRS; it flashes and transmits, but max output power is about -10dbm.

**Comment by uberhalit** [comment]
> Sorry, no time to finish, anyone may use this as a reference.

@lesykm do you remember what parts are missing for a functional release? I got a RadioMaster Bandit Nano which uses the same ESP32-D0WDQ6 + SX1276 combo as your TX. I ported over some code from ELRS; it flashes and transmits, but max output power is about -10dbm.

**Pull request closed without merging**